juriscraper/opinions/united_states/administrative_agency/bia.py

```diff
@@ -0,0 +1,75 @@
+"""Scraper for Dept of Justice. Board of Immigration Appeals
+CourtID: bia
+Court Short Name: Dept of Justice. Board of Immigration Appeals
+Author: William Palin
+Reviewer:
+Type:
+History:
+    2021-12-18: Created by William E. Palin
+"""
+
+from typing import Any, Dict
+
+from juriscraper.OpinionSiteLinear import OpinionSiteLinear
+
+
+class Site(OpinionSiteLinear):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.court_id = self.__module__
+        self.url = "https://www.justice.gov/eoir/ag-bia-decisions"
+        self.article = None
+
+    def _download(self, request_dict={}):
+        html = super()._download(request_dict)
+        self.url = html.xpath(".//table//tbody/tr/td/a/@href")[0]
+        if self.test_mode_enabled():
+            return html
+        return self._get_html_tree_by_url(self.url)
+
+    def process_elements(self, elements) -> Dict[str, Any]:
```

Comment:
This method needs a lot more comments to explain things. Can you try to think of future-Bill and how you'd make him happy and add comments accordingly?

Reply:
fair.  I was expecting to have to trash this whole thing but ... let me see what I can do.